### PR TITLE
Update opentelemetry dependency

### PR DESCRIPTION
**Pull request by joefreeman**
## Description

This updates the (optional) `opentelemetry` dependency from `0.17b0` to `1.4.1`.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Joe Freeman for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

File: strawberry/extensions/tracing/opentelemetry.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ async def resolve(self, _next, root, info, *args, **kwargs):
 
             return result
 
-        with self._tracer.use_span(self._root_span):
+        with trace.use_span(self._root_span):
```
Comment (patrick91) [inline strawberry/extensions/tracing/opentelemetry.py:79]: why was this changed? 😊

Comment (joefreeman) [reply strawberry/extensions/tracing/opentelemetry.py:79]: Looks like it got moved in open-telemetry/opentelemetry-python#1668 :shrug:

Comment (patrick91) [reply strawberry/extensions/tracing/opentelemetry.py:79]: gotcha! thanks for explaing! 😊

File: pyproject.toml [hunk 2]
```diff
@@ -59,8 +59,8 @@ requests = "^2.26.0"
 pre-commit = "^2.14.0"
 pytest-benchmark = "^3.4.1"
 freezegun = "^1.1.0"
-opentelemetry-api = "^0.17b0"
-opentelemetry-sdk = "^0.17b0"
+opentelemetry-api = "^1.4.1"
+opentelemetry-sdk = "^1.4.1"
```
Comment (joefreeman) [inline pyproject.toml:63]: I guess this might be an annoying upgrade for anyone who is using `opentelemetry==0.17b0`?

Comment (patrick91) [reply pyproject.toml:63]: it could be, maybe we can handle both versions? we only have two places where we had to change our code, so maybe we can do it

**Comment by joefreeman** [comment]
> it could be, maybe we can handle both versions? we only have two places where we had to change our code, so maybe we can do it

Something like this?

I'm not sure how to specify the version to allow the `0.17b0` pre-release, but prevent from using pre-releases after `1.4.1` (i.e., `1.10a0`). Any ideas?

**Comment by patrick91** [comment]
> I'm not sure how to specify the version to allow the `0.17b0` pre-release, but prevent from using pre-releases after `1.4.1` (i.e., `1.10a0`). Any ideas?

mhm, we could allow any version before 2? `<2`?

**Comment by joefreeman** [comment]
> mhm, we could allow any version before 2? `<2`?

I think that still installs `1.10a0` though. Maybe not a big deal, but I think I've found what I was looking for: `"^0.17b0 || ~1.4"` - I think that requires either the specific version we want to keep for backwards compatibility, or something 1.4ish.

**Comment by patrick91** [comment]
@joefreeman sounds good to me 😊 thanks!

**Review by patrick91 — APPROVED** [review]
💯
